httpclient5/src/main/java/org/apache/hc/client5/http/entity/DecompressEntity.java

```diff
@@ -0,0 +1,160 @@
+     */
+    private InputStream content;
+
+    /**
+     * The compression type used for decompression (e.g., gzip, deflate).
+     */
+    private final String compressionType;
+
+    /**
+     * The flag indicating if decompression should skip certain headers (nowrap).
+     */
+    private final boolean nowrap;
+
+    /**
+     * Constructs a new {@link DecompressEntity} with the specified compression type and nowrap setting.
+     *
+     * @param wrapped         the non-null {@link HttpEntity} to be wrapped.
+     * @param compressionType the compression type (e.g., "gzip", "deflate").
+     * @param nowrap          whether to decompress without headers for certain compression formats.
+     */
+    public DecompressEntity(final HttpEntity wrapped, final String compressionType, final boolean nowrap) {
+        super(wrapped);
+        this.compressionType = compressionType;
+        this.nowrap = nowrap;
+    }
+
+    /**
+     * Constructs a new {@link DecompressEntity} with the specified compression type, defaulting to no nowrap handling.
+     *
+     * @param wrapped         the non-null {@link HttpEntity} to be wrapped.
+     * @param compressionType the compression type (e.g., "gzip", "deflate").
+     */
+    public DecompressEntity(final HttpEntity wrapped, final String compressionType) {
+        this(wrapped, compressionType, false);
+    }
+
+    /**
+     * Initializes and returns a stream for decompression.
+     * The decompression is applied lazily on the wrapped entity's content.
+     *
+     * @return a lazily initialized {@link InputStream} that decompresses the content.
+     * @throws IOException if an error occurs during decompression.
+     */
+    private InputStream getDecompressingStream() throws IOException {
+        return new LazyDecompressInputStream(super.getContent(), compressionType, nowrap);
+    }
+
+    /**
+     * Returns the decompressed content stream. If the entity is streaming,
+     * the same {@link InputStream} is returned on subsequent calls.
+     *
+     * @return the decompressed {@link InputStream}.
+     * @throws IOException if an error occurs during decompression.
+     */
+    @Override
+    public InputStream getContent() throws IOException {
+        if (super.isStreaming()) {
+            if (content == null) {
+                content = getDecompressingStream();
+            }
+            return content;
+        }
+        return getDecompressingStream();
+    }
+
+    /**
+     * Writes the decompressed content to the specified {@link OutputStream}.
+     *
+     * @param outStream the {@link OutputStream} to which the decompressed content is written; must not be {@code null}.
+     * @throws IOException if an I/O error occurs during writing or decompression.
+     */
+    @Override
+    public void writeTo(final OutputStream outStream) throws IOException {
+        Args.notNull(outStream, "Output stream");
```

Comment:
You can reuse Commons IO's `IOUtils.copy(InputStream, OutputStream) which Commons Compress uses already.

Reply:
HI @garydgregory  
That method is deprecated
`/**
     * Copies the content of a InputStream into an OutputStream. Uses a default buffer size of 8024 bytes.
     *
     * @param input  the InputStream to copy
     * @param output the target, may be null to simulate output to dev/null on Linux and NUL on Windows
     * @return the number of bytes copied
     * @throws IOException if an error occurs
     * @deprecated Use {@link org.apache.commons.io.IOUtils#copy(InputStream, OutputStream)}.
     */
    @Deprecated
    public static long copy(final InputStream input, final OutputStream output) throws IOException {
        return org.apache.commons.io.IOUtils.copy(input, output);
    }`

Reply:
Hello @arturobernalg 
It is not deprecated. You're confusing Commons IO IOUtils and the Commons Compress IOUtils, which delegates to Commons IO.